portal-bridge/src/bridge/history.rs

```diff
@@ -175,49 +176,38 @@ impl HistoryBridge {
                         Please specify a starting block/epoch that begins before the current block."
             );
         }
-        let current_epoch = latest_block / EPOCH_SIZE;
-        let mut gossip_range = Range {
+        let mut current_epoch_index = start_block / EPOCH_SIZE;
+        let gossip_range = Range {
             start: start_block,
             end: end_block,
         };
-        while epoch_index <= current_epoch {
+
+        info!("fetching headers in range: {gossip_range:?}");
+        let mut epoch_acc = None;
+        for height in gossip_range {
             // Using epoch_size chunks & epoch boundaries ensures that every
             // "chunk" shares an epoch accumulator avoiding the need to
             // look up the epoch acc on a header by header basis
-            let epoch_acc = if gossip_range.end <= MERGE_BLOCK_NUMBER {
-                match self.get_epoch_acc(epoch_index).await {
+            if height <= MERGE_BLOCK_NUMBER && current_epoch_index != height / EPOCH_SIZE {
```

Comment:
I don't think this is correct. At least on the first pass through. Let's say `current_epoch_index = 4`. On the first loop, we won't set the `epoch_acc` correctly, since the second clause here is not `true`. I don't think the `epoch_acc` will be set correctly for any blocks in the first epoch, but once we hit the next epoch, and the second clause evaluates to `true` then it should work

Reply:
Good catch!

A solution for this I just pushed was to
```
        // initialize current_epoch_index as an impossible value u64::MAX so that
        // epoch_acc gets set on the first iteration of the loop
        let mut current_epoch_index = u64::MAX;
```

Set current_epoch_index to MAX as that should be an impossible value and now ``epoch_acc`` will be properly set on the first iteration of the loop